### PR TITLE
Add Eureka service discovery documentation

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
I didn't add the documentation about Eureka.

Modifications:
- Add server-service-registration
- Remove advanced-zookeeper and move the contents to server-service-registration and client-service-discovery

### CODE REVIEW

**Comment by minwoox** [comment]
server-service-registration

**Comment by minwoox** [comment]
client-service-discovery

File: site/src/pages/docs/client-service-discovery.mdx [hunk 1]
```diff
@@ -234,4 +234,76 @@ group.whenReady().get();
...
+                              .build();
+```
+
+The <type://ZooKeeperEndpointGroup> is used to fetch the binary representation of server information.
+The <type://ZooKeeperDiscoverySpec> converts the binary representation to an <type://Endpoint>.
+
+<type://ZooKeeperDiscoverySpec#curator(String)> uses the format of
+[Curator Service Discovery](https://curator.apache.org/curator-x-discovery/index.html). If you registered your
+server with <type://ZooKeeperRegistrationSpec#curator(String)>, you must use
+<type://ZooKeeperDiscoverySpec#curator(String)>.
```
[the remark this answers is not in the document]
Comment (minwoox) [reply site/src/pages/docs/client-service-discovery.mdx:259]: Fixed. 😉

File: site/src/pages/docs/setup.mdx [hunk 2]
```diff
@@ -68,10 +72,15 @@ the list of major Armeria artifacts which might interest you:
 |                                             |  See [Embedding a servlet container](/docs/server-servlet).                     |
 +---------------------------------------------+---------------------------------------------------------------------------------+
 | `armeria-zookeeper3`                        | ZooKeeper 3 based service discovery.                                            |
-|                                             | See [Service discovery with ZooKeeper](/docs/advanced-zookeeper).               |
+|                                             | See [ZooKeeper service discovery][ZooKeeper service discovery] and              |
```
[the remark this answers is not in the document]
Comment (minwoox) [reply site/src/pages/docs/setup.mdx]: Fixed. 😉

File: site/src/pages/docs/client-service-discovery.mdx [hunk 3]
```diff
@@ -234,4 +234,77 @@ group.whenReady().get();
...
+ZooKeeperEndpointGroup myEndpointGroup =
+        ZooKeeperEndpointGroup.builder(client, znodePath, discoverySpec)
+                              .build();
+```
+
+For more information, please refer to the API documentation of the
+[com.linecorp.armeria.client.zookeeper](https://javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/latest/com/linecorp/armeria/server/client/package-summary.html) package.
+
+### See also
+- [ZooKeeper service registration](/docs/server-service-registration#zookeeper-based-service-registration-with-zookeeperupdatinglistener)
```
Comment (trustin) [inline site/src/pages/docs/client-service-discovery.mdx:280]: How about using `<Tip />` ?

File: site/src/pages/docs/client-service-discovery.mdx [hunk 4]
```diff
@@ -234,4 +234,77 @@ group.whenReady().get();
...
+builder.regions("aws")
+       .appName("myApplication");
+EurekaEndpointGroup eurekaEndpointGroup = builder.build();
+```
+
+For more information, please refer to the API documentation of the
+[com.linecorp.armeria.client.eureka](https://javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/latest/com/linecorp/armeria/client/eureka/package-summary.html) package.
+
+### See also
+- [Eureka service registration](/docs/server-service-registration#eureka-based-service-registration-with-eurekaupdatinglistener)
```
Comment (trustin) [inline site/src/pages/docs/client-service-discovery.mdx:310]: Ditto

File: site/src/pages/docs/server-service-registration.mdx [hunk 5]
```diff
@@ -1,6 +1,16 @@
-# Service discovery with ZooKeeper
+# Service registration for discovery
 
-## Automatic service registration
+You can use <type://ServerListener> to register the information of Armeria servers when they start so that
+a client uses the information to distribute its requests to the servers autonomously, unlike
+traditional server-side load balancing where the requests go through a dedicated load balancer such as
+[L4 and L7 switches](https://en.wikipedia.org/wiki/Multilayer_switch#Layer_4%E2%80%937_switch,_web_switch,_or_content_switch).
+
+Armeria provides 2 <type://ServerListener> implementations for registering:
```
Comment (trustin) [inline site/src/pages/docs/server-service-registration.mdx:8]: for service registration:

File: site/src/pages/docs/server-service-registration.mdx [hunk 6]
```diff
@@ -58,55 +68,51 @@ server.start();
...
-        ZooKeeperEndpointGroup.builder(zkConnectionStr, znodePath, discoverySpec)
-                              .sessionTimeoutMillis(10000)
-                              .build();
-```
+[com.linecorp.armeria.server.zookeeper](https://javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/latest/com/linecorp/armeria/server/zookeeper/package-summary.html) package.
 
-The <type://ZooKeeperEndpointGroup> is used to fetch the binary representation of server information.
-The <type://ZooKeeperDiscoverySpec> converts the binary representation to an <type://Endpoint>.
+### See also
+- [ZooKeeper service discovery](/docs/client-service-discovery#zookeeper-based-service-discovery-with-zookeeperendpointgroup)
```
Comment (trustin) [inline site/src/pages/docs/server-service-registration.mdx:74]: `<Tip />` ?

File: site/src/pages/docs/server-service-registration.mdx [hunk 7]
```diff
@@ -58,55 +68,51 @@ server.start();
...
 
 ```java
-import com.linecorp.armeria.common.SessionProtocol;
-
-WebClient webClient = WebClient.of(SessionProtocol.HTTP, myEndpointGroup, "/hello");
+import com.linecorp.armeria.server.eureka.EurekaUpdatingListenerBuilder;
+
+EurekaUpdatingListenerBuilder builder =
+        EurekaUpdatingListener.builder("https://eureka.com:8001/eureka/v2");
+builder.instanceId("i-00000000")
```
Comment (trustin) [inline site/src/pages/docs/server-service-registration.mdx:101]: Maybe you can just chain?

**Comment by minwoox** [comment]
@trustin All fixed. PTAL 😉

**Review by trustin — APPROVED** [review]
Thanks, @minwoox !

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox

**Pull request merged**